featurebyte/worker/task/catalog_cleanup.py

```diff
@@ -0,0 +1,230 @@
+                pass
+
+    async def _cleanup_store_files(self, remote_file_paths: set[Path]) -> None:
+        for remote_file_path in remote_file_paths:
+            await self.storage.try_delete_if_exists(remote_file_path)
+
+    async def _cleanup_catalog(self, catalog: CatalogModel) -> None:
+        query_filter = {"catalog_id": catalog.id}
+        for collection_name, model_class in self.catalog_specific_model_class_pairs:
+            docs = await self.persistent.get_iterator(
+                collection_name=collection_name,
+                query_filter=query_filter,
+            )
+
+            # extract the warehouse tables and remote file paths of the model collection
+            warehouse_tables = set()
+            remote_file_paths = set()
+            async for doc_dict in docs:
+                obj = model_class(**doc_dict)
+                assert isinstance(obj, FeatureByteCatalogBaseDocumentModel)
+                warehouse_tables.update(obj.warehouse_tables)
+                remote_file_paths.update(obj.remote_storage_paths)
+
+            # cleanup the warehouse tables & remote files
+            await self._cleanup_warehouse_tables(
+                catalog=catalog,
+                warehouse_tables=warehouse_tables,
+            )
+            await self._cleanup_store_files(remote_file_paths)
+
+            # delete the mongo documents
+            await self.persistent.delete_many(
+                collection_name=model_class.collection_name(),
+                query_filter=query_filter,
+                user_id=catalog.user_id,
+            )
+
+        # cleanup user defined functions
+        await self.persistent.delete_many(
+            collection_name=UserDefinedFunctionModel.collection_name(),
+            query_filter={"catalog_id": catalog.id},
+            user_id=catalog.user_id,
+        )
+
+        # cleanup task documents
+        await self.persistent.delete_many(
+            collection_name=Task.collection_name(),
+            query_filter={"kwargs.catalog_id": str(catalog.id)},
+            user_id=catalog.user_id,
+        )
+
+        # cleanup the catalog document
+        await self.all_catalog_service.delete_document(
+            document_id=catalog.id,
+            user_id=catalog.user_id,
+            use_raw_query_filter=True,
+        )
+
+    async def execute(self, payload: CatalogCleanupTaskPayload) -> Any:
+        # compute the cutoff time
+        cutoff_time = datetime.now() - timedelta(days=payload.cleanup_threshold_in_days)
+        with self.all_catalog_service.allow_use_raw_query_filter():
+            async for catalog in self.all_catalog_service.list_documents_iterator(
+                query_filter={
+                    "is_deleted": True,
+                    "updated_at": {"$lt": cutoff_time},
+                },
+                use_raw_query_filter=True,
+            ):
+                await self._cleanup_catalog(catalog)
```

Comment:
Would it be useful to wrap this in a generic try catch block to handle unexpected errors so that one error doesn't block the other catalogs from being cleaned up?